compiler/rustc_borrowck/src/used_muts.rs

```diff
@@ -66,8 +66,8 @@ impl<'visit, 'cx, 'tcx> Visitor<'tcx> for GatherUsedMutsVisitor<'visit, 'cx, 'tc
     fn visit_terminator(&mut self, terminator: &Terminator<'tcx>, location: Location) {
         debug!("visit_terminator: terminator={:?}", terminator);
         match &terminator.kind {
-            TerminatorKind::Call { destination: Some((into, _)), .. } => {
-                self.remove_never_initialized_mut_locals(*into);
+            TerminatorKind::Call { destination, target: Some(_), .. } => {
```

Comment:
I _think_ this should be able to operate correctly regardless of `target`? 
```suggestion
            TerminatorKind::Call { destination, .. } => {
```